### PR TITLE
Add support for moderation report notifications

### PR DESCRIPTION
**Pull request by Tak**
Clicking either avatar takes you to the appropriate profile, clicking the notification opens the report link in the browser

Tusky:
Screen Shot 2022-11-21 at 14 44 47

Web:
Screen Shot 2022-11-21 at 15 00 04

(`violation`/`spam` is the literal category field from the server, not sure whether it makes sense to add prettified/translated descriptions etc.)

closes https://github.com/tuskyapp/Tusky/issues/2933

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+class ReportNotificationViewHolder(
+    private val binding: ItemReportNotificationBinding,
+) : RecyclerView.ViewHolder(binding.root) {
+
+    fun setupWithReport(reporter: TimelineAccount, report: Report, animateAvatar: Boolean, animateEmojis: Boolean) {
+        val reporterName = reporter.name.unicodeWrap().emojify(reporter.emojis, itemView, animateEmojis)
+        val reporteeName = report.targetAccount.name.unicodeWrap().emojify(report.targetAccount.emojis, itemView, animateEmojis)
+        val icon = ContextCompat.getDrawable(itemView.context, R.drawable.ic_flag_24dp)?.apply {
+            setColorFilter(itemView.context.getColor(R.color.tusky_blue), PorterDuff.Mode.SRC_ATOP)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt:45]: put the color inside the drawable xml and remove the color filter please

File: app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            val position = bindingAdapterPosition
+            if (position != RecyclerView.NO_POSITION) {
+                listener.onViewAccount(reporterId)
+            }
+        }
+
+        itemView.setOnClickListener { listener.onViewReport(reportId) }
+    }
+
+    private fun getIconWithColor(
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt:87]: unused

File: app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            setColorFilter(itemView.context.getColor(R.color.tusky_blue), PorterDuff.Mode.SRC_ATOP)
+        }
+        binding.notificationTopText.setCompoundDrawablesWithIntrinsicBounds(icon, null, null, null)
+        binding.notificationTopText.text = itemView.context.getString(R.string.notification_header_report_format, reporterName, reporteeName)
+        binding.notificationSummary.text = itemView.context.getString(R.string.notification_summary_report_format, TimestampUtils.getRelativeTimeSpanString(itemView.context, report.createdAt.time, Date().time), report.status_ids?.size ?: 0)
+        binding.notificationCategory.text = report.category
+
+        // Fancy avatar inset
+        val padding = Utils.dpToPx(binding.notificationReporteeAvatar.context, 12)
+        binding.notificationReporteeAvatar.setPaddingRelative(0, 0, padding, padding)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt:54]: Can't this be in the layout xml?

**Review by connyduck — COMMENTED** [review]
> (violation/spam is the literal category field from the server, not sure whether it makes sense to add prettified/translated descriptions etc.)

I think it is better if it is translateable. [The possible categories seem to be violation/spam/other.](https://docs.joinmastodon.org/entities/Report/) Ideally, fallback to printing the name in case they introduce new categories.

File: app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt (hunk 3 above)
Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt:54]: I tried, but I was never able to get it to work?
The other notification and status views are doing it this way as well…

Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt:49]: ok, let's leave it then

**Review by connyduck — APPROVED** [review]

**Pull request merged**